TAScheduler/views_methods.py

```diff
@@ -61,45 +64,288 @@ def createSection(self, section_info):
         pass
 
     def removeCourse(self, active_course):
-        pass
+        if type(active_course) is not CourseObj:
+            raise TypeError("Input passed is not a Course object")
+        elif not Course.objects.filter(course_id=active_course.database.course_id).exists():
+            raise RuntimeError("Course does not exist")
+        Course.delete(active_course.database)
 
     def removeUser(self, active_user):
-        pass
+        if not isinstance(active_user, UserObj):
+            raise TypeError("Input passed is not a subclass of userobj")
+        elif not User.objects.filter(email_address=active_user.getUsername()).exists():
+            raise RuntimeError("User does not exist")
+        User.delete(active_user.database.user)
 
     def removeSection(self, active_section):
-        pass
+        if not isinstance(active_section, SectionObj):
+            raise TypeError("Input passed is not a subclass of sectionobj")
+        elif not Section.objects.filter(section_id=active_section.getID()).exists():
+            raise RuntimeError("Section does not exist")
+        Section.delete(active_section.database.section)
 
     def editCourse(self, active_course, new_info):  # new inputs
-        pass
+        if type(active_course) is not CourseObj:
+            raise TypeError("Input passed is not a Course object")
+        elif not Course.objects.filter(course_id=active_course.database.course_id).exists():
+            raise RuntimeError("Course does not exist")
+        if type(new_info) is not dict:
+            raise TypeError("Input passed is not a dictionary")
+
+        try:  # Course ID
+            if new_info.get("course_id") is None:
+                raise KeyError
+            if type(new_info.get("course_id")) is not int or new_info.get("course_id") < 0:
+                raise ValueError("Course id expects an int")
+            if Course.objects.filter(course_id=new_info.get("course_id")).exists():
+                raise RuntimeError("Can not have two courses with the same course number")
+            active_course.database.course_id = new_info.get("course_id")
+        except KeyError:  # No course_id in list that is fine don't change the database
+            active_course.database.course_id = active_course.database.course_id
```

Comment:
So if there are any exceptions in the try block, it will get caught in the except block ... why would we not just want to  throw an exception to the caller (like lines 88-93)? If the ID is invalid then why would we still want to keep looking at the other fields?

Reply:
Because if this is checking the new_info that was passed. If course_id is not passed that is fine it is not being updated. We can still update other portions of the course